content/blog/2016/2016-11-16-blueocean-beta-declarative-pipeline-pipeline-editor.adoc

```diff
@@ -0,0 +1,127 @@
+
+=== Whats included?
+
+link:/blog/2016/05/26/introducing-blue-ocean/[Back in April we open sourced Blue Ocean]
+and shared our vision with the community. We're very happy that all the things we showed you then have
+shipped in the beta (software projects run *on time*?!).
+
+For a refresher on Blue Ocean, watch this short video:
+++++
+<center>
+<iframe width="853" height="480"
+src="https://www.youtube-nocookie.com/embed/3dITffteCD4?rel=0" frameborder="0"
+allowfullscreen></iframe>
+</center>
+++++
+
+=== Declarative Pipeline
+
+We heard a lot from the community about the usability of link:/doc/pipeline[Jenkins
+Pipeline]. Much of the feedback we received was boiled down to a desire to
+*configure* Pipelines rather than *script* them and to make it easy for beginners
+to get started with their first Pipeline.
+
+And that is how *Declarative Pipeline* was born. We've introduced a new method whereby
+you declare how you want your Pipeline to look rather than using Pipeline Script
+ - its configuration rather than code.
+
+Here's a small example of a Declarative Pipeline for nodejs that runs the whole
+Pipeline inside a Docker container:
+```
+pipeline {
+  agent docker:'node:6.3'
+  stages {
+    stage('build') {
+      sh 'npm --version'
+      sh 'npm install'
+    }
+    stage ('test') {
+      sh 'npm test'
+    }
+  }
+}
+```
+
+Docker support in Declarative Pipeline allows you to version your application code,
+Jenkins Pipeline configuration and the environment where your pipeline will run,
+all in a single repository. It's a crazy powerful combination.
+
+Declarative pipeline also introduces the `postBuild` section that makes it
+easy to run things conditionally at the end of your Pipeline without the
+complexity of the `try... catch` of Pipeline script.
+
+```
+postBuild {
+  always {
+    sh 'echo "This will always run"'
+  }
+  success {
+    sh 'echo "This will run only if successful"'
+  }
+  failure {
+    sh 'echo "This will run only if failed"'
+  }
+  unstable {
+    sh 'echo "This will run only if the run was marked as unstable"'
+  }
+  changed {
+    sh 'echo "This will run only if the state of the Pipeline has changed"'
+    sh 'echo "For example, the Pipeline was previously failing but is now successful"'
+    sh 'echo "... or the other way around :)"'
+  }
+}
+```
+
+and there is so much more!
+
+If you have the Blue Ocean beta installed you already have Declarative Pipeline.
+While Declarative Pipeline is still alpha at the moment, we do encourage you to
+https://github.com/jenkinsci/pipeline-model-definition-plugin/wiki/getting%20started[follow our getting started guide]
+ and give us feedback on the link:/content/mailing-lists/[Jenkins Users mailing list].
+
+=== Introducing the Pipeline Editor
+
+The Pipeline Editor is a graphical user interface that gives Jenkins users the
+simplest way to get started with creating Pipelines in Jenkins. It will also
```

Comment:
Another personal wording preference - "the simplest way yet". Because y'never know when we might get even simpler.